packages/web/src/components/RegistrationModal/RegistrationForm.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import {
+  Input,
+  FormControl,
+  VStack,
+  FormHelperText,
+  FormLabel,
+  Box,
+  Button,
+  Textarea,
+  InputProps,
+  TextareaProps,
+} from '@chakra-ui/react';
+import { AlertResponse } from './AlertResponse';
+import { useRegistrationConfig } from './useRegistrationConfig';
+import { RegistrationFormProps, FormProps, RegistrationSchema, RegistrationFormik } from './utils';
+import { Hint } from './HintTooltip';
+import { statusColors } from '../../theme/colors';
+
+const handleBlur = async (e: React.FocusEvent<HTMLInputElement>, formik: RegistrationFormik) => {
+  const values = {
+    [e.target.id]: e.target.value,
+    // if any outstanding validation errors exist, add them to be validated
+    ...Object.entries(formik.errors)
+      .map(([key]) => {
+        const value = formik.values[key as keyof RegistrationSchema];
+        return { [key]: value };
+      })
+      .reduce((acc, curr) => ({ ...acc, ...curr }), {}),
+  };
+  await formik.validateForm(values);
+};
+const formProps: FormProps = (key, formik) => ({
+  variant: 'filled',
+  type: 'text',
+  value: formik.values[key],
+  isInvalid: !!formik.errors[key],
+  onChange: formik.handleChange,
+  onBlur: (e: React.FocusEvent<HTMLInputElement>) => handleBlur(e, formik),
+});
+
+export const RegistrationForm: React.FC<RegistrationFormProps> = ({ initialValues, onSubmit }) => {
+  const { formik, alertDescription, alertProps } = useRegistrationConfig({
+    initialValues,
+    onSubmit,
+  });
+  return (
+    <Box>
+      <form onSubmit={formik.handleSubmit}>
+        <VStack alignItems="stretch">
+          <FormControl id="name">
+            <FormLabel>
+              Name
+              <Hint> A descriptive title for your app </Hint>
+            </FormLabel>
+            <Input {...(formProps('name', formik) as InputProps)} />
```

Comment:
Was the assertion here needed? If so let's chat because a better way to do this would be to add a generic to the type for `formProps`

Reply:
In the future we can look at generics